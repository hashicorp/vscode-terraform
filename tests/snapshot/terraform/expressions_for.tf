[for s in var.list : upper(s)]

[for k, v in var.map : length(k) + length(v)]

[for i, v in var.list : "${i} is ${v}"]

{for s in var.list : s => upper(s)}

[for s in var.list : upper(s) if s != ""]

locals {
  admin_users = {
    for name, user in var.users : name => user if user.is_admin
  }
  regular_users = {
    for name, user in var.users : name => user
    if !user.is_admin
  }
  admin_users_list = [
    for name, user in var.users : name if user.is_admin
  ]
  regular_users_list = [
    for name, user in var.users : name
    if !user.is_admin
  ]
}

locals {
  users_by_role = {
    for name, user in var.users : user.role => name...
  }
}

