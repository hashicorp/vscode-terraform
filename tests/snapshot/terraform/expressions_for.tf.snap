>[for s in var.list : upper(s)]
#^ source.terraform punctuation.section.brackets.begin.terraform
# ^^^ source.terraform keyword.control.terraform
#    ^ source.terraform
#     ^ source.terraform variable.other.readwrite.terraform
#      ^ source.terraform
#       ^^ source.terraform keyword.operator.word.terraform
#         ^ source.terraform
#          ^^^ source.terraform support.constant.terraform
#             ^ source.terraform keyword.operator.accessor.terraform
#              ^^^^ source.terraform variable.other.member.terraform
#                  ^ source.terraform
#                   ^ source.terraform keyword.operator.terraform
#                    ^ source.terraform
#                     ^^^^^ source.terraform meta.function-call.terraform support.function.builtin.terraform
#                          ^ source.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                           ^ source.terraform meta.function-call.terraform
#                            ^ source.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#                             ^ source.terraform punctuation.section.brackets.end.terraform
>
>[for k, v in var.map : length(k) + length(v)]
#^ source.terraform punctuation.section.brackets.begin.terraform
# ^^^ source.terraform keyword.control.terraform
#    ^ source.terraform
#     ^ source.terraform variable.other.readwrite.terraform
#      ^ source.terraform punctuation.separator.terraform
#       ^ source.terraform
#        ^ source.terraform variable.other.readwrite.terraform
#         ^ source.terraform
#          ^^ source.terraform keyword.operator.word.terraform
#            ^ source.terraform
#             ^^^ source.terraform support.constant.terraform
#                ^ source.terraform keyword.operator.accessor.terraform
#                 ^^^ source.terraform variable.other.member.terraform
#                    ^ source.terraform
#                     ^ source.terraform keyword.operator.terraform
#                      ^ source.terraform
#                       ^^^^^^ source.terraform meta.function-call.terraform support.function.builtin.terraform
#                             ^ source.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                              ^ source.terraform meta.function-call.terraform
#                               ^ source.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#                                ^ source.terraform
#                                 ^ source.terraform keyword.operator.arithmetic.terraform
#                                  ^ source.terraform
#                                   ^^^^^^ source.terraform meta.function-call.terraform support.function.builtin.terraform
#                                         ^ source.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                                          ^ source.terraform meta.function-call.terraform
#                                           ^ source.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#                                            ^ source.terraform punctuation.section.brackets.end.terraform
>
>[for i, v in var.list : "${i} is ${v}"]
#^ source.terraform punctuation.section.brackets.begin.terraform
# ^^^ source.terraform keyword.control.terraform
#    ^ source.terraform
#     ^ source.terraform variable.other.readwrite.terraform
#      ^ source.terraform punctuation.separator.terraform
#       ^ source.terraform
#        ^ source.terraform variable.other.readwrite.terraform
#         ^ source.terraform
#          ^^ source.terraform keyword.operator.word.terraform
#            ^ source.terraform
#             ^^^ source.terraform support.constant.terraform
#                ^ source.terraform keyword.operator.accessor.terraform
#                 ^^^^ source.terraform variable.other.member.terraform
#                     ^ source.terraform
#                      ^ source.terraform keyword.operator.terraform
#                       ^ source.terraform
#                        ^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                         ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#                           ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#                            ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
#                             ^^^ source.terraform string.quoted.double.terraform
#                                ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform
#                                 ^^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.begin.terraform
#                                   ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform variable.other.readwrite.terraform
#                                    ^ source.terraform string.quoted.double.terraform meta.interpolation.terraform keyword.other.interpolation.end.terraform
#                                     ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                                      ^ source.terraform punctuation.section.brackets.end.terraform
>
>{for s in var.list : s => upper(s)}
#^ source.terraform punctuation.section.braces.begin.terraform
# ^^^ source.terraform keyword.control.terraform
#    ^ source.terraform
#     ^ source.terraform variable.other.readwrite.terraform
#      ^ source.terraform
#       ^^ source.terraform keyword.operator.word.terraform
#         ^ source.terraform
#          ^^^ source.terraform support.constant.terraform
#             ^ source.terraform keyword.operator.accessor.terraform
#              ^^^^ source.terraform variable.other.member.terraform
#                  ^ source.terraform
#                   ^ source.terraform keyword.operator.terraform
#                    ^ source.terraform
#                     ^ source.terraform variable.other.readwrite.terraform
#                      ^ source.terraform
#                       ^^ source.terraform storage.type.function.terraform
#                         ^ source.terraform
#                          ^^^^^ source.terraform meta.function-call.terraform support.function.builtin.terraform
#                               ^ source.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                                ^ source.terraform meta.function-call.terraform
#                                 ^ source.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#                                  ^ source.terraform punctuation.section.braces.end.terraform
>
>[for s in var.list : upper(s) if s != ""]
#^ source.terraform punctuation.section.brackets.begin.terraform
# ^^^ source.terraform keyword.control.terraform
#    ^ source.terraform
#     ^ source.terraform variable.other.readwrite.terraform
#      ^ source.terraform
#       ^^ source.terraform keyword.operator.word.terraform
#         ^ source.terraform
#          ^^^ source.terraform support.constant.terraform
#             ^ source.terraform keyword.operator.accessor.terraform
#              ^^^^ source.terraform variable.other.member.terraform
#                  ^ source.terraform
#                   ^ source.terraform keyword.operator.terraform
#                    ^ source.terraform
#                     ^^^^^ source.terraform meta.function-call.terraform support.function.builtin.terraform
#                          ^ source.terraform meta.function-call.terraform punctuation.section.parens.begin.terraform
#                           ^ source.terraform meta.function-call.terraform
#                            ^ source.terraform meta.function-call.terraform punctuation.section.parens.end.terraform
#                             ^ source.terraform
#                              ^^ source.terraform keyword.control.conditional.terraform
#                                ^ source.terraform
#                                 ^ source.terraform variable.other.readwrite.terraform
#                                  ^ source.terraform
#                                   ^^ source.terraform keyword.operator.terraform
#                                     ^ source.terraform
#                                      ^ source.terraform string.quoted.double.terraform punctuation.definition.string.begin.terraform
#                                       ^ source.terraform string.quoted.double.terraform punctuation.definition.string.end.terraform
#                                        ^ source.terraform punctuation.section.brackets.end.terraform
>
>locals {
#^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#      ^ source.terraform meta.block.terraform
#       ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  admin_users = {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#             ^ source.terraform meta.block.terraform variable.declaration.terraform
#              ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform
#                ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.begin.terraform
>    for name, user in var.users : name => user if user.is_admin
#^^^^ source.terraform meta.block.terraform meta.braces.terraform
#    ^^^ source.terraform meta.block.terraform meta.braces.terraform keyword.control.terraform
#       ^ source.terraform meta.block.terraform meta.braces.terraform
#        ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.separator.terraform
#             ^ source.terraform meta.block.terraform meta.braces.terraform
#              ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform meta.braces.terraform
#                   ^^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.word.terraform
#                     ^ source.terraform meta.block.terraform meta.braces.terraform
#                      ^^^ source.terraform meta.block.terraform meta.braces.terraform support.constant.terraform
#                         ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.accessor.terraform
#                          ^^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.member.terraform
#                               ^ source.terraform meta.block.terraform meta.braces.terraform
#                                ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.terraform
#                                 ^ source.terraform meta.block.terraform meta.braces.terraform
#                                  ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                                      ^ source.terraform meta.block.terraform meta.braces.terraform
#                                       ^^ source.terraform meta.block.terraform meta.braces.terraform storage.type.function.terraform
#                                         ^ source.terraform meta.block.terraform meta.braces.terraform
#                                          ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                                              ^ source.terraform meta.block.terraform meta.braces.terraform
#                                               ^^ source.terraform meta.block.terraform meta.braces.terraform keyword.control.conditional.terraform
#                                                 ^ source.terraform meta.block.terraform meta.braces.terraform
#                                                  ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                                                      ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.accessor.terraform
#                                                       ^^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.member.terraform
>  }
#^^ source.terraform meta.block.terraform meta.braces.terraform
#  ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.end.terraform
>  regular_users = {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform
#                ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                 ^ source.terraform meta.block.terraform variable.declaration.terraform
#                  ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.begin.terraform
>    for name, user in var.users : name => user
#^^^^ source.terraform meta.block.terraform meta.braces.terraform
#    ^^^ source.terraform meta.block.terraform meta.braces.terraform keyword.control.terraform
#       ^ source.terraform meta.block.terraform meta.braces.terraform
#        ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.separator.terraform
#             ^ source.terraform meta.block.terraform meta.braces.terraform
#              ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform meta.braces.terraform
#                   ^^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.word.terraform
#                     ^ source.terraform meta.block.terraform meta.braces.terraform
#                      ^^^ source.terraform meta.block.terraform meta.braces.terraform support.constant.terraform
#                         ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.accessor.terraform
#                          ^^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.member.terraform
#                               ^ source.terraform meta.block.terraform meta.braces.terraform
#                                ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.terraform
#                                 ^ source.terraform meta.block.terraform meta.braces.terraform
#                                  ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                                      ^ source.terraform meta.block.terraform meta.braces.terraform
#                                       ^^ source.terraform meta.block.terraform meta.braces.terraform storage.type.function.terraform
#                                         ^ source.terraform meta.block.terraform meta.braces.terraform
#                                          ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
>    if !user.is_admin
#^^^^ source.terraform meta.block.terraform meta.braces.terraform
#    ^^ source.terraform meta.block.terraform meta.braces.terraform keyword.control.conditional.terraform
#      ^ source.terraform meta.block.terraform meta.braces.terraform
#       ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.logical.terraform
#        ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.accessor.terraform
#             ^^^^^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.member.terraform
>  }
#^^ source.terraform meta.block.terraform meta.braces.terraform
#  ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.end.terraform
>  admin_users_list = [
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform variable.declaration.terraform
#                   ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                    ^ source.terraform meta.block.terraform variable.declaration.terraform
#                     ^ source.terraform meta.block.terraform punctuation.section.brackets.begin.terraform
>    for name, user in var.users : name if user.is_admin
#^^^^ source.terraform meta.block.terraform
#    ^^^ source.terraform meta.block.terraform keyword.control.terraform
#       ^ source.terraform meta.block.terraform
#        ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform punctuation.separator.terraform
#             ^ source.terraform meta.block.terraform
#              ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform
#                   ^^ source.terraform meta.block.terraform keyword.operator.word.terraform
#                     ^ source.terraform meta.block.terraform
#                      ^^^ source.terraform meta.block.terraform support.constant.terraform
#                         ^ source.terraform meta.block.terraform keyword.operator.accessor.terraform
#                          ^^^^^ source.terraform meta.block.terraform variable.other.member.terraform
#                               ^ source.terraform meta.block.terraform
#                                ^ source.terraform meta.block.terraform keyword.operator.terraform
#                                 ^ source.terraform meta.block.terraform
#                                  ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
#                                      ^ source.terraform meta.block.terraform
#                                       ^^ source.terraform meta.block.terraform keyword.control.conditional.terraform
#                                         ^ source.terraform meta.block.terraform
#                                          ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
#                                              ^ source.terraform meta.block.terraform keyword.operator.accessor.terraform
#                                               ^^^^^^^^ source.terraform meta.block.terraform variable.other.member.terraform
>  ]
#^^ source.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform punctuation.section.brackets.end.terraform
>  regular_users_list = [
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#                    ^ source.terraform meta.block.terraform variable.declaration.terraform
#                     ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                      ^ source.terraform meta.block.terraform variable.declaration.terraform
#                       ^ source.terraform meta.block.terraform punctuation.section.brackets.begin.terraform
>    for name, user in var.users : name
#^^^^ source.terraform meta.block.terraform
#    ^^^ source.terraform meta.block.terraform keyword.control.terraform
#       ^ source.terraform meta.block.terraform
#        ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform punctuation.separator.terraform
#             ^ source.terraform meta.block.terraform
#              ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform
#                   ^^ source.terraform meta.block.terraform keyword.operator.word.terraform
#                     ^ source.terraform meta.block.terraform
#                      ^^^ source.terraform meta.block.terraform support.constant.terraform
#                         ^ source.terraform meta.block.terraform keyword.operator.accessor.terraform
#                          ^^^^^ source.terraform meta.block.terraform variable.other.member.terraform
#                               ^ source.terraform meta.block.terraform
#                                ^ source.terraform meta.block.terraform keyword.operator.terraform
#                                 ^ source.terraform meta.block.terraform
#                                  ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
>    if !user.is_admin
#^^^^ source.terraform meta.block.terraform
#    ^^ source.terraform meta.block.terraform keyword.control.conditional.terraform
#      ^ source.terraform meta.block.terraform
#       ^ source.terraform meta.block.terraform keyword.operator.logical.terraform
#        ^^^^ source.terraform meta.block.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform keyword.operator.accessor.terraform
#             ^^^^^^^^ source.terraform meta.block.terraform variable.other.member.terraform
>  ]
#^^ source.terraform meta.block.terraform
#  ^ source.terraform meta.block.terraform punctuation.section.brackets.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>locals {
#^^^^^^ source.terraform meta.block.terraform entity.name.type.terraform
#      ^ source.terraform meta.block.terraform
#       ^ source.terraform meta.block.terraform punctuation.section.block.begin.terraform
>  users_by_role = {
#^^ source.terraform meta.block.terraform
#  ^^^^^^^^^^^^^ source.terraform meta.block.terraform variable.declaration.terraform variable.other.readwrite.terraform
#               ^ source.terraform meta.block.terraform variable.declaration.terraform
#                ^ source.terraform meta.block.terraform variable.declaration.terraform keyword.operator.assignment.terraform
#                 ^ source.terraform meta.block.terraform variable.declaration.terraform
#                  ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.begin.terraform
>    for name, user in var.users : user.role => name...
#^^^^ source.terraform meta.block.terraform meta.braces.terraform
#    ^^^ source.terraform meta.block.terraform meta.braces.terraform keyword.control.terraform
#       ^ source.terraform meta.block.terraform meta.braces.terraform
#        ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#            ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.separator.terraform
#             ^ source.terraform meta.block.terraform meta.braces.terraform
#              ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                  ^ source.terraform meta.block.terraform meta.braces.terraform
#                   ^^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.word.terraform
#                     ^ source.terraform meta.block.terraform meta.braces.terraform
#                      ^^^ source.terraform meta.block.terraform meta.braces.terraform support.constant.terraform
#                         ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.accessor.terraform
#                          ^^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.member.terraform
#                               ^ source.terraform meta.block.terraform meta.braces.terraform
#                                ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.terraform
#                                 ^ source.terraform meta.block.terraform meta.braces.terraform
#                                  ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                                      ^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.accessor.terraform
#                                       ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.member.terraform
#                                           ^ source.terraform meta.block.terraform meta.braces.terraform
#                                            ^^ source.terraform meta.block.terraform meta.braces.terraform storage.type.function.terraform
#                                              ^ source.terraform meta.block.terraform meta.braces.terraform
#                                               ^^^^ source.terraform meta.block.terraform meta.braces.terraform variable.other.readwrite.terraform
#                                                   ^^^ source.terraform meta.block.terraform meta.braces.terraform keyword.operator.terraform
>  }
#^^ source.terraform meta.block.terraform meta.braces.terraform
#  ^ source.terraform meta.block.terraform meta.braces.terraform punctuation.section.braces.end.terraform
>}
#^ source.terraform meta.block.terraform punctuation.section.block.end.terraform
>
>